lib/src/checks.dart

```diff
@@ -305,54 +306,58 @@ class GuildCheck extends Check {
 }
 
 /// Represents different types of cooldown
-enum CooldownType {
+class CooldownType extends IEnum<int> {
   /// Cooldown is per category.
   ///
   /// If the command is executed in a guild channel belonging to a category, the cooldown is set for
   /// all users in all channels belonging to that category.
   ///
   /// If the channel does not belong to a category or is not a guild channel, the cooldown works in
   /// the same way as [channel].
-  category,
+  static const CooldownType category = CooldownType(1 << 0);
```

Comment:
Why use bitshifting now just a single number e.g. 1, 2, 3

Reply:
For consistency with the other constructor calls, I think the readability is worth more than the actual utility of the code here.

Reply:
And if it was a single number, it would have to be 1 as a binary or is used to combine cooldown types.